Admin.js

```diff
@@ -114,561 +136,771 @@ var Admin = {
-                success: function(response) {
-                    var elm = jQuery(subject).parent();
+                "url": subject.attr("href"),
+                "type": "POST",
+                "success" (response) {
+
+                    const elm = jQuery(subject).parent();
                     elm.children().remove();
-                    // fix issue with html comment ...
+                    // Fix issue with html comment ...
                     elm.html(jQuery(response.replace(/<!--[\s\S]*?-->/g, "")).html());
-                    elm.effect("highlight", {'color' : '#57A957'}, 2000);
+                    elm.effect("highlight", {"color": "#57A957"}, 2000);
                     Admin.set_object_field_value(elm);
+
                 },
-                error: function(xhr, statusText, errorThrown) {
-                    jQuery(subject).parent().effect("highlight", {'color' : '#C43C35'}, 2000);
+                "error" (xhr, statusText, errorThrown) {
+
+                    jQuery(subject).parent().
+                        effect("highlight", {"color": "#C43C35"}, 2000);
+
                 }
             });
+
         });
+
     },
 
-    setup_collection_counter: function(subject) {
-        Admin.log('[core|setup_collection_counter] setup collection counter', subject);
+    "setup_collection_counter" (subject) {
+
+        Admin.log("[core|setup_collection_counter] setup collection counter", subject);
 
         // Count and save element of each collection
-        var highestCounterRegexp = new RegExp('_([0-9]+)[^0-9]*$');
-        jQuery(subject).find('[data-prototype]').each(function() {
-            var collection = jQuery(this);
-            var counter = 0;
-            collection.children().each(function() {
-                var matches = highestCounterRegexp.exec(jQuery('[id^="sonata-ba-field-container"]', this).attr('id'));
-                if (matches && matches[1] && matches[1] > counter) {
-                    counter = parseInt(matches[1], 10);
-                }
+        const highestCounterRegexp = new RegExp("_([0-9]+)[^0-9]*$");
+        jQuery(subject).find("[data-prototype]").
+            each(function () {
+
+                const collection = jQuery(this);
+                let counter = 0;
+                collection.children().each(function () {
+
+                    const matches = highestCounterRegexp.exec(jQuery("[id^=\"sonata-ba-field-container\"]", this).attr("id"));
+                    if (matches && matches[1] && matches[1] > counter) {
+
+                        counter = parseInt(matches[1], 10);
+
+                    }
+
+                });
+                Admin.lolCounters[collection.attr("id")] = counter;
+
             });
-            Admin.lolCounters[collection.attr('id')] = counter;
-        });
+
     },
 
-    setup_collection_buttons: function(subject) {
+    "setup_collection_buttons" (subject) {
+
+        jQuery(subject).on("click", ".sonata-collection-add", function (event) {
```

Comment:
- eslint: Unexpected unnamed function.
- eslint: Function has too many statements (12). Maximum allowed is 10.